packages/react-noop-renderer/src/createReactNoop.js

```diff
@@ -770,7 +775,12 @@ function createReactNoop(reconciler: Function, useMutation: boolean) {
 
       // flush effects until none remain, and cleanup
       try {
-        if (actingUpdatesScopeDepth === 1) {
+        if (
+          actingUpdatesScopeDepth === 1 &&
+          (isSchedulerMocked === false ||
+            (isSchedulerMocked === true &&
```

Comment:
`isSchedulerMocked === true` always evaluates to true here because `isSchedulerMocked === false` already evaluated to false.